tests/integration/actions/exec/base.py

```diff
@@ -0,0 +1,104 @@
+
+    UPDATE_FIXTURES = False
+    PANE_HEIGHT = 25
+    PANE_WIDTH = 300
+    CONFIG_FILE: Union[str, None] = None
+
+    @pytest.fixture(scope="module", name="tmux_session")
+    def fixture_tmux_session(
+        self, request: pytest.FixtureRequest
+    ) -> Generator[TmuxSession, None, None]:
+        """Tmux fixture for this module.
+
+        :param request: The request for this fixture
+        :yields: A tmux session
+        """
+        params = {
+            "unique_test_id": request.node.nodeid,
+            "pane_height": self.PANE_HEIGHT,
+            "pane_width": self.PANE_WIDTH,
+        }
+        if isinstance(self.CONFIG_FILE, str):
+            assert os.path.exists(self.CONFIG_FILE)
+            params["config_path"] = self.CONFIG_FILE
+
+        with TmuxSession(**params) as tmux_session:
+            yield tmux_session
+
+    def test(self, request: pytest.FixtureRequest, tmux_session: TmuxSession, step: Step):
+        # pylint:disable=unused-argument
+        # pylint: disable=too-few-public-methods
+        # pylint: disable=too-many-arguments
+        """Test interactive/stdout exec.
+
+        :param request: The test request
+        :param tmux_session: The tmux session
+        :param step: A step within a series of tests
+        :raises ValueError: If test mode isn't set
+        """
+        if step.search_within_response is SearchFor.PROMPT:
+            search_within_response = tmux_session.cli_prompt
+        else:
+            raise ValueError("test mode not set")
+
+        received_output = tmux_session.interaction(
+            value=step.user_input,
+            search_within_response=search_within_response,
+        )
+
+        if (
+            self.UPDATE_FIXTURES
+            or os.environ.get("ANSIBLE_NAVIGATOR_UPDATE_TEST_FIXTURES") == "true"
+        ):
+            update_fixtures(
+                request,
+                step.step_index,
+                received_output,
+                step.comment,
+                additional_information={
+                    "look_fors": step.look_fors,
+                    "look_nots": step.look_nots,
+                    "compared_fixture": not any((step.look_fors, step.look_nots)),
+                },
+            )
+
+        page = " ".join(received_output)
+
+        if step.look_fors:
+            assert all(look_for in page for look_for in step.look_fors)
+
+        if step.look_nots:
+            assert not any(look_not in page for look_not in step.look_nots)
+
+        if not any((step.look_fors, step.look_nots)):
+            dir_path, file_name = fixture_path_from_request(request, step.step_index)
+            with open(f"{dir_path}/{file_name}") as infile:
+                expected_output = json.load(infile)["output"]
+
+            assert expected_output == received_output, "\n" + "\n".join(
+                difflib.unified_diff(expected_output, received_output, "expected", "received")
+            )
```

Comment:
This is a nice opportunity to dedent a code block:
```suggestion
        if any((step.look_fors, step.look_nots)):
            return

        dir_path, file_name = fixture_path_from_request(request, step.step_index)
        with open(f"{dir_path}/{file_name}") as infile:
            expected_output = json.load(infile)["output"]

        assert expected_output == received_output, "\n" + "\n".join(
            difflib.unified_diff(expected_output, received_output, "expected", "received")
        )
```

Reply:
interesting....  in this case I prefer the first because it's explicit.  "What happens if there are not any...."  in the update that becomes implied by the early return, which suggests an invisible else in there. That said, opportunities to dedent should be watched for, but I'm gonna leave this one as-is for now.